src/test/kotlin/io/github/mojira/arisa/ModuleRegistryTest.kt

```diff
@@ -0,0 +1,56 @@
+package io.github.mojira.arisa
+
+import com.uchuhimo.konf.Config
+import io.github.mojira.arisa.infrastructure.config.Arisa
+import io.github.mojira.arisa.modules.Module
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.collections.shouldContainExactlyInAnyOrder
+import io.kotest.matchers.shouldBe
+import org.reflections.Reflections
+
+class ModuleRegistryTest : StringSpec({
+    "should register a module for each config class" {
+        val modules = ModuleRegistry(getConfig()).getModules().map { it.name }
+        val configs = Arisa.Modules::class.java.classes.map { it.simpleName }.filter { !it.endsWith("Spec") }
+        println("Configs not mapped to a registered module " + configs.filter { !modules.contains(it) })
+        println("Registered modules not mapped to a config " + modules.filter { !configs.contains(it) })
+        configs shouldContainExactlyInAnyOrder modules
+    }
+
+    "should register max 1 time each config" {
+        val modules = ModuleRegistry(getConfig()).getModules().map { it.config.prefix }
+        val uniqueModules = modules.distinct()
+        println("Not unique modules " + modules.filter { !uniqueModules.contains(it) })
+        uniqueModules shouldContainExactlyInAnyOrder modules
+    }
+
+    "should get only get a module with only" {
+        val modules = ModuleRegistry(getConfig(true)).getModules()
+        modules.size shouldBe 1
+    }
+
+    "should register a module for each file ending in Module" {
```

Comment:
[Minor] Maybe change the description to `should register a module for each non-abstract module`, to also cover Abstract...?
(Also not sure if you can use reflections to look for abstract, maybe we don't need to rely on the file name)